pkg/filesystem/referer_adaptor.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package filesystem
+
+import (
+	"context"
+	"fmt"
+
+	snpkg "github.com/containerd/containerd/pkg/snapshotters"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+func (fs *Filesystem) CheckReferrer(ctx context.Context, labels map[string]string) bool {
+	if !fs.ReferrerDetectEnabled() {
```

Comment:
It's better not to bind Referrer manger  with Filesystem since filesystem is a very thin layer to adapt to snapshots layer

Reply:
It seems `cacheMgr` and `stargzResolver` still also exist here..

Reply:
Fine, I am planning to refactor them. Let's keep it the same way now.